src/graph.jl

```diff
@@ -0,0 +1,74 @@
+using JSON
+using SparseArrays
+using LightGraphs
+abstract type AbstractGraph end
+
+struct Graph<:AbstractGraph
+    num_nodes::Int
+    num_edges::Int
+
+    # an adjacency matrix where each entry is the edge id that connects
+    # the two nodes.
+    adj_matrix::SparseMatrixCSC{Int, Int}
+
+    # source and destination of the edge. the index of these arrays are the
+    # edge ids.
+    edge_src::Array{Int, 1}
+    edge_dst::Array{Int, 1}
+
+    # an Array of Arrays where the neighbors of node `i` are found in an array
+    # at index `i`
+    node_neighbors::Array{Array{Int64,1},1}
```

Comment:
-> `neighbors`?

Reply:
ok: node_neighbors -> neighbors, edge_src -> src, edge_dst-> dst

Reply:
running into a problem: the lightgraphs library has a function called src() and dst(), and this is causing namespace issues. In the Graph function we want to use the src()/dst() functions, and multiple dispatch doesnt work on data structures i dont think. Leaving it as edge_src for now but I agree this is kinda ugly. Will think more.